### DISCUSSION TITLE
caprover using the wrong IP for docker swarm

### DISCUSSION THREAD

**Discussion by jonathonbyrdziak** [caprover/caprover#1364] [Q&A]
I'm setting up caprover on aws ec2, but I'm unable to use docker swarm.

caprover is forcing the swarm init on docker run. This presents a problem when we're unable to specify docker swarm options, like "advertise-addr".

I attempted to boo the swarm first, and then docker run, but caprover was stuck in a reboot loop until I left the swarm.

How can I pass the proper ip address to caprover?

**Comment by githubsaturn** [comment]
You can use `MAIN_NODE_IP_ADDRESS` env variable during installation.

example:
```
docker run -e MAIN_NODE_IP_ADDRESS=192.168.1.12 -p 80:80 -p 443:443 -p 3000:3000 -v /var/run/docker.sock:/var/run/docker.sock -v /captain:/captain caprover/caprover
```
